src/libcore/mem.rs

```diff
@@ -483,6 +483,15 @@ pub fn align_of_val<T: ?Sized>(val: &T) -> usize {
 /// ```
 #[inline]
 #[stable(feature = "needs_drop", since = "1.21.0")]
+#[cfg(not(stage0))]
+pub const fn needs_drop<T>() -> bool {
+    intrinsics::needs_drop::<T>()
```

Comment:
I’m surprised `unsafe` block is not necessary here.

Reply:
That's because I made it safe (see the changes to `src/librustc_typeck/check/intrinsic.rs` and `src/librustc_typeck/collect.rs`). Not doing so results in the following error when building:

```
error: call to unsafe function is unsafe and unsafe operations are not allowed in const fn
```

Reply:
And indeed `needs_drop` is always safe to call so this is fine.